### PR TITLE
Feature/operational panels backend

### PR DESCRIPTION
**Pull request by ps48**
### Description
Added Backend APIs to the Operational Panels using Observability plugin

### Issues Resolved
1. Added Operational Panels Kibana APIs #99 #100 
2. Event Explorer API endpoint for adding visualizations to any panel

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/constants/shared.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export const observabilityPluginOrder = 6000;
 export const UI_DATE_FORMAT = 'MM/DD/YYYY hh:mm A';
 export const PPL_DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss';
 export const PPL_INDEX_REGEX = /(search source|source|index)\s*=\s*([^|\s]+)/i;
+export const PPL_CONTAINS_TIMESTAMP_REGEX = /\|\s*.*\s*[<|<=|=|>=|>]\s*timestamp\([^\)]+\)/i;
```
Comment (mengweieric) [inline common/constants/shared.ts:33]: Do you need this timestamp regex in custom panel? This was added previously but the new requirement asks to not check if a query contains a date range anymore, I've already removed on my side.

**Review by mengweieric — APPROVED** [review]

File: server/adaptors/custom_panels/custom_panel_adaptor.ts [hunk 2]
```diff
@@ -50,44 +50,335 @@ export class CustomPanelsAdaptor {
...
+        objectType: 'operationalPanel',
+      });
+      return response.observabilityObjectList.map((panel: any) => ({
+        name: panel.operationalPanel.name,
+        id: panel.objectId,
+        dateCreated: panel.createdTimeMs,
+        dateModified: panel.lastUpdatedTimeMs,
       }));
     } catch (error) {
       if (error.body.error.type === 'index_not_found_exception') {
```
Comment (joshuali925) [inline server/adaptors/custom_panels/custom_panel_adaptor.ts:78]: you can remove this condition, the backend will handle it

File: server/adaptors/custom_panels/custom_panel_adaptor.ts [hunk 3]
```diff
@@ -50,44 +50,335 @@ export class CustomPanelsAdaptor {
...
+      });
+      return response.observabilityObjectList.map((visualization: any) => ({
+        id: visualization.objectId,
+        name: visualization.savedVisualization.name,
+        query: this.savedVisualizationsQueryBuilder(visualization.savedVisualization.query),
+        type: visualization.savedVisualization.type,
+        timeField: visualization.savedVisualization.selected_timestamp.name,
+      }));
+    } catch (error) {
+      if (error.body.error.type === 'index_not_found_exception') {
```
Comment (joshuali925) [inline server/adaptors/custom_panels/custom_panel_adaptor.ts:224]: same here

File: server/routes/custom_panels/panels_router.ts [hunk 4]
```diff
@@ -415,16 +275,26 @@ export function PanelsRouter(router: IRouter) {
...
+        );
         return response.ok({
           body: {
-            message: 'Panel DateTime Filter Changed',
+            message: 'Panel PPL Filter Changed',
           },
         });
       } catch (error) {
-        console.log('Issue in adding datetime filter', error);
+        console.log('Issue in adding query filter', error);
```
Comment (joshuali925) [inline server/routes/custom_panels/panels_router.ts:297]: maybe use `console.error`?

File: common/constants/shared.ts (hunk 1 above)
Comment (ps48) [reply common/constants/shared.ts:33]: This is added here cause, I still need to remove time range from queries in "savedVisualization" data modal. 
1. If we are still allowing the users to add time range in query in event analytics, then I need to remove them explicitly in panels' visualizations.
2. If we are not allowing users to add time range in query in event analytics, then I'll remove this check. 
Please let me know if got something wrong here.

Comment (ps48) [reply common/constants/shared.ts:33]: Oh yes, I understand this point that date range is stored separately and only added in runtime. But, I would like to know what happens if the user adds a time range in the query? Do we explicitly remove? Or Do we throw an error toast saying "adding time range is not allowed in query"? If we have such checks I'll remove the checks from my side.

**Pull request merged**